openlibrary/plugins/upstream/checkins.py

```diff
@@ -81,14 +75,17 @@ def POST(self, work_id):
         edition_key : string : optional
         event_id : int : optional
         """
-        data = json.loads(web.data())
-
-        if not self.validate_data(data):
-            raise web.badrequest(message='Invalid date submitted')
-
         user = get_current_user()
         if not user:
-            raise web.unauthorized(message='Requires login')
+            raise web.HTTPError(
+                "401 Unauthorized", headers={"Content-Type": "application/json"}
+            )
+
+        data = json.loads(web.data())
+        if not self.validate_data(data):
+            raise web.HTTPError(
+                "404 Bad Request", headers={"Content-Type": "application/json"}
```

Comment:
```suggestion
                "404 Not Found", headers={"Content-Type": "application/json"}
```
Fix copy-paste oversight.

Reply:
This might be best as a 400 Bad Request than a 404